projects/storefrontlib/src/lib/cms-lib/product-carousel/product-carousel.component.ts

```diff
@@ -42,11 +24,9 @@ export class ProductCarouselComponent implements OnInit {
   private window: Window;
 
   constructor(
-    public component: CmsComponentData<CmsProductCarouselComponent>,
-    private productService: ProductService,
     winRef: WindowRef,
     private el: ElementRef,
-    private cd: ChangeDetectorRef
+    private productCarouselService: ProductCarouselService
```

Comment:
probably we can use now: private service: ProductCarouselService

Reply:
it seems too generic for me, what if we would add any other service in the future?

Reply:
Kacper comment same in my PR https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/1140#discussion_r253806566
This refactor cause there will be no any other services in future.

Reply:
done